src/devices/machine/upd765.cpp

```diff
@@ -425,6 +434,25 @@ uint8_t ps2_fdc_device::sra_r()
 
 uint8_t ps2_fdc_device::srb_r()
 {
+ 	/*
+	 *  b7: always 1
+	 *	b6: always 1
+	 *	b5: drive 0 select
+	 *	b4: write data
+	 *	b3: read data
+	 *	b2: write enable
+	 *	b1: motor enable 1
+	 *	b0: motor enable 0
+	 */
+
+	uint8_t data = 0xC0;
+
+	data |= (flopi[0].dev && flopi[0].dev->mon_r()) ? 0x00 : 0x01;
+	data |= (flopi[1].dev && flopi[1].dev->mon_r()) ? 0x00 : 0x02;
+	data |= (flopi[0].dev && flopi[0].dev->ds_r()) ? 0x20 : 0x00;
+
+	return data;
+	/*
```

Comment:
Are you sure this is what you want to do here?  Commenting out the previous implementation entirely looks a bit dangerous.  Also, MAME convention is to use lowercase hex digits.

Reply:
I'm working on the implementation, but the standard uPD SRB byte is incorrect for the PS/2.